src/script/interpreter.cpp

```diff
@@ -282,15 +285,16 @@ bool EvalScript(vector<vector<unsigned char> >& stack, const CScript& script, un
             // Read instruction
             //
             if (!script.GetOp(pc, opcode, vchPushValue))
-                return set_error(serror, SCRIPT_ERR_BAD_OPCODE);
-            if (vchPushValue.size() > MAX_SCRIPT_ELEMENT_SIZE)
+                return protocol_cleanup ? true : set_error(serror, SCRIPT_ERR_BAD_OPCODE);
```

Comment:
You need to return `set_success(serror)`, not `true` here.

I also think you should throw an error unless this has been mined into a block. I.e. it should be a policy rule to not allow unknown opcodes, but not a consensus rule. (I noticed you added the policy cleanup flag to the mandatory flags, without which you would have actually gotten the opposite outcome here (policy = allow, consensus = forbid).)

In other words, perhaps there needs to be a second `VERIFY_BAD_OPCODES` flag (kept as `verify_badopcodes` below) which would turn this into
```suggestion
                return protocol_cleanup && !verify_badopcodes ? set_success(serror) : set_error(serror, SCRIPT_ERR_BAD_OPCODE);
```
It would be non-mandatory policy-only, just like e.g. `SCRIPT_VERIFY_CHECKSEQUENCEVERIFY` and friends (i.e. it would be in `STANDARD_SCRIPT_VERIFY_FLAGS`).

Reply:
I think `SCRIPT_VERIFY_DISCOURAGE_UPGRADABLE_NOPS` can be reused for this purpose?

Reply:
Done, using `SCRIPT_VERIFY_DISCOURAGE_UPGRADABLE_NOPS` instead.

Reply:
Ahh yeah, that works!